datafusion/physical-expr/src/aggregate/utils.rs

```diff
@@ -211,3 +211,21 @@ pub(crate) fn ordering_fields(
 pub fn get_sort_options(ordering_req: &[PhysicalSortExpr]) -> Vec<SortOptions> {
     ordering_req.iter().map(|item| item.options).collect()
 }
+
+/// A wrapper around a type to provide hash for primitive arrow types
+#[derive(Copy, Clone)]
+pub(crate) struct Hashable<T>(pub T);
+
+impl<T: ToByteSlice> std::hash::Hash for Hashable<T> {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.0.to_byte_slice().hash(state)
```

Comment:
I think this is only needed for float types? It would probably be faster to use the native hash implementation for other (integer) types?

Reply:
You're right, it only required for floats (and it only makes sense to use it for floats, cause default implementations of these traits for integers outperforms this one).

Reply:
It makes sense to have an implementation for primitive and float types instead one using `ToByteSlice` then?